doc/source/serve/deployment-graph.md

```diff
@@ -0,0 +1,401 @@
+
+We want to leverage the programmable and general purpose distributed computing ability of Ray, double down on its unique strengths (scheduling, communication and shared memory) to facilitate authoring, orchestrating, scaling and deployment of complex serving pipelines under one set of DAG API, so a user can program & test multiple models or multiple shards of a single large model dynamically, deploy to production at scale, and upgrade individually.
+
+## Key requirements
+- Provide the ability to author a DAG of Serve nodes to form a complex inference graph.
+- Pipeline authoring experience should be fully Python-programmable with support for dynamic selection, control flows, user business logic, etc.
+- DAG can be instantiated and locally executed using tasks and actors API
+- DAG can be deployed as a group where individual nodes can be reconfigured and scaled indepenently.
+
+__[Full Ray Enhancement Proposal, REP-001: Serve Pipeline](https://github.com/ray-project/enhancements/blob/main/reps/2022-03-08-serve_pipeline.md)__
+
++++
+
+## Concepts
+
+### Deployment
+Upgradeable group of actors managed by the Serve controller. Currently the primary API in Ray Serve. At authoring time it’s the class or function under `@serve.deployment` decorator. 
+
+### Node
+Smallest unit in a graph, typically an annotated class but can also be a function, backed by a group of actors that are scalable and reconfigurable. 
+
+### Deployment Graph
+Collection of nodes that forms a DAG that represents an inference graph for complicated tasks. Ex: ensemble, chaining, dynamic selection. 
+
+### Bind
+A graph building API applicable to decorated class or function.  `decorated_class_or_func.bind(*args, **kwargs)` generates an IR node that can be used to build graph, and bound arguments will be applied at execution time, including dynamic user input.
+
++++
+
+## Key APIs Explained
+
+The class and function definition as well as decorator didn't diverage from existing serve API. So in the following section we only need to dive into a few new key APIs used: `bind()`, `InputNode()` and `serve.run()`
+
++++
+
+### **`bind(*args, **kwargs)`**
+
+Once called on supported ray decorated function or class (@ray.remote, @serve.deployment), generates an IR of type DAGNode that acts as the building block of graph building.
+
++++
+
+#### On function
+
+```bind()``` on function produces a DAGNode that can be exeucted with user input.
+
+```{code-cell} ipython3
+import ray
+from ray import serve
+from ray.serve.pipeline.generate import DeploymentNameGenerator
+
+if ray.is_initialized():
+    serve.shutdown()
+    DeploymentNameGenerator.reset()
+    ray.shutdown()
+
+ray.init(num_cpus=4)
+serve.start()
```

Comment:
do we need all of this boilerplate?

Reply:
can't users just do `serve run` repeatedly?

Reply:
explained below